### PR TITLE
ጪ Correctly embed JS in templates

### PR DESCRIPTION
**Pull request by dodobas**
The correct way to embed JS in templates is to use `escapejs` and
`JSON.parse` to read data.

### CODE REVIEW

File: temba/utils/templatetags/temba.py [hunk 1]
```diff
@@ -150,3 +152,21 @@ def render(self, context):
 
 # register our tag
 lessblock = register.tag(lessblock)
+
+
+@register.filter
+def to_js(value):
```
Comment (rowanseymour) [inline temba/utils/templatetags/temba.py]: minor point but I'd call this `to_json` for clarity

File: temba/utils/tests.py [hunk 2]
```diff
@@ -491,6 +493,24 @@ def forloop(idx, total):
...
         self.assertEqual(".", oxford(forloop(3, 4), "."))
 
+    def test_to_js(self):
+        from temba.utils.templatetags.temba import to_js
+
+        # only works with plain str objects
+        self.assertRaises(ValueError, to_js, dict())
+
+        self.assertEqual(to_js(json.dumps({})), 'JSON.parse("{}")')
+        self.assertEqual(to_js(json.dumps({"a": 1})), 'JSON.parse("{\\u0022a\\u0022: 1}")')
```
Comment (rowanseymour) [inline temba/utils/tests.py]: does kinda suck from a debugging pov that you now won't be able to read these so easily in the page source, but more important to deal with potential XSS exploits

File: temba/utils/templatetags/temba.py [hunk 3]
```diff
@@ -150,3 +152,21 @@ def render(self, context):
...
+    """
+    To use a python variable in JS, we call json.dumps to serialize as JSON server-side and reconstruct using
+    JSON.parse. The serialized string must be escaped appropriately before dumping into the client-side code.
+
+    https://stackoverflow.com/a/14290542
+    """
+    # separators is passed to remove whitespace in output
```
Comment (rowanseymour) [inline temba/utils/templatetags/temba.py]: this comment refers to the use of the `separators` arg to `escapejs` which you're not using

**Comment by dodobas** [comment]
The first implementation was doing just that, but I didn't like it because we would remove 'markers' from views.py which hint that that 'context variable' is somehow special. When I read code I like to know upfront that a context variable is 'different' for any reason, in stead of hunting it down through templates...

**Comment by rowanseymour** [comment]
@dodobas personally I'd prefer we did that via naming (e.g. `foo_json` is something to be rendered as JSON) and left the thing in the context decoded so that you can easily check it in tests etc, but no biggie.

**Pull request merged**